Code/Tools/AssetProcessor/Platform/Mac/native/FileWatcher/FileWatcher_macos.cpp

```diff
@@ -60,44 +36,47 @@ bool FolderRootWatch::Start()
     // Set ourselves as the value for the context info field so that in the callback
     // we get passed into it and the callback can call our public API to handle
     // the file change events
-    FSEventStreamContext streamContext;
-    ::memset(&streamContext, 0, sizeof(streamContext));
-    streamContext.info = this;
+    FSEventStreamContext streamContext{
+        /*.version =*/ 0,
+        /*.info =*/ this,
+    };
 
-    m_platformImpl->m_stream = FSEventStreamCreate(NULL,
+    m_platformImpl->m_stream = FSEventStreamCreate(nullptr,
                                  FileEventStreamCallback,
                                  &streamContext,
                                  pathsToWatch,
                                  kFSEventStreamEventIdSinceNow,
                                  timeBetweenKernelUpdateAndNotification,
                                  kFSEventStreamCreateFlagFileEvents);
 
-    AZ_Error("FileWatcher", (m_platformImpl->m_stream != nullptr), "FSEventStreamCreate returned a nullptr. No file events will be reported for %s", m_root.toStdString().c_str());
-
-    m_thread = std::thread(std::bind(&FolderRootWatch::WatchFolderLoop, this));
+    AZ_Error("FileWatcher", (m_platformImpl->m_stream != nullptr), "FSEventStreamCreate returned a nullptr. No file events will be reported.");
 
+    const CFIndex pathCount = CFArrayGetCount(pathsToWatch);
+    for(CFIndex i = 0; i < pathCount; ++i)
+    {
+        CFRelease(CFArrayGetValueAtIndex(pathsToWatch, i));
+    }
     CFRelease(pathsToWatch);
-    CFRelease(rootPath);
 
-    return (m_platformImpl->m_stream != nullptr);
+    return m_platformImpl->m_stream != nullptr;
 }
 
-void FolderRootWatch::Stop()
+void FileWatcher::PlatformStop()
 {
     m_shutdownThreadSignal = true;
 
     if (m_thread.joinable())
     {
         m_thread.join(); // wait for the thread to finish
-        m_thread = std::thread(); //destroy
+        m_thread = AZStd::thread(); //destroy
```

Comment:
This was removed for the linux implementation, but kept here?

Reply:
Fixed.